Dockerfile

```diff
@@ -12,19 +12,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG GOARCH="amd64"
-
-FROM golang:1.24 AS builder
-# golang envs
-ARG GOARCH="amd64"
-ARG GOOS=linux
-ENV CGO_ENABLED=0
+# setup cross-compile env
+FROM --platform=$BUILDPLATFORM golang:1.24 AS builder
+ARG TARGETARCH
+ENV GOARCH=${TARGETARCH} CGO_ENABLED=0
```

Comment:
Normally I would use ARG instead of env for build-time values, but since this is an intermediate build step it doesn't matter and this is cleaner.